Source/Targeting.swift

```diff
@@ -133,6 +133,27 @@ import CoreLocation
         }
     }
     
+    public var purposeConsents: String? {
+        set {
+            StorageUtils.setPurposeConsents(value: newValue)
+        }
+
+        get {
+            var savedConsent: String?
+            
+            if let pbString = StorageUtils.pbPurposeConsents() {
+                savedConsent = pbString
+            } else if let iabString = StorageUtils.iabPurposeConsents() {
+                savedConsent = iabString
+            }
+            if(savedConsent != nil){
```

Comment:
can we move this logic from getter to `RequestBuilder` class ?

Reply:
I need to check if the value is set by the publisher or are we accessing from the user preference. and the string that i extract from consent string, I only need the 1st char which represents the purposeConsents